### ISSUE TITLE
Some use cases for the return of making some redirects be optionally 302

### ISSUE THREAD

**Issue by MadtownLems** [markjaquith/page-links-to#120]
Hiya,

In our brief Twitter exchange, you asked why we were hoping to see the return of an option to make some redirects be 302 (temporary) instead of 301 (permanent).

I can present three specific cases that happened in our department recently, but it really just comes down to the fact that sometimes you only want to send visitors somewhere specific as a temporary measure - or WHERE you want to send them is temporary.  Once a 301 is "out there" in browsers (and google), you lose control over it and these can be cached for incredibly long periods of time (browsers typically cache them, as the name says, permanently - until its cache is full or is cleared).

1) One of our websites recently had the embedded contact form on their (very active) contact page stop working, so the site administrators used Page Links To to redirect those visitors to an offsite form.  When we replaced their embedded contact form and removed the Page Links To redirect a couple of weeks later, many people were unable to access it due to their browsers' caches.

2) On another site that was being soft-launched, the site builder had the following situation: a top-level "Overview Page" with child pages for various modules.  At SOME point, they expect to populate the "Overview Page", but it's not ready yet. So they'd like to temporarily redirect anyone who visits "Overview" to just end up at "Overview Page -> Module 1" instead until they find the time to actually finish up their page.

3) We are in the middle of moving quite a few things around. An educator used Page Links To in a pretty traditional way, but the destination she was linking TO was going to change in a couple of weeks. A 302 would have been much more appropriate here.

I fully understand the desire to keep options to a minimum, keep the UI as clean as possible, and keep maintenance down - but I think there are enough use cases for 302 redirects that they would be a valuable addition.

**Comment by markjaquith** [comment]
Yep, you make a good points! I might implement this as hidden-by-default but make it opt-in, the same way you can opt out of the "new tab" feature and its UI. I'm loath to add a settings screen to the plugin, and also to clutter the UI too much by default. I did a survey and people overwhelmingly didn't want the core plugin to become too heavy.

**Comment by MadtownLems** [comment]
Totally agree with loathing a settings page. In our environment, I have written many a custom-plugin to hide the settings page for plugins. ("You needed a custom, top-level settings page to choose the color of part of the UI, really?")

Happy to help test or provide feedback before/after release. You know where to reach me. :)

Thanks again for maintaining the plugin - we use it a lot!
